[v3] Document type inference with import.meta.glob

### Describe the bug

When using `import.meta.glob` with vite v3, no generic type is provided anymore. The [JS Doc provides you with this information](https://github.com/vitejs/vite/blob/main/packages/vite/types/importGlob.d.ts#L41), but it would have helped me to have it in the [official docs](https://vitejs.dev/guide/features.html#glob-import) easily available.

Also, is this not a breaking change for typescript users, strictly speaking?

Thanks for a great project.

### Reproduction

https://stackblitz.com/edit/vitejs-vite-lgkx3h?file=src/main.ts&view=editor

### System Info

```shell
System:
    OS: Linux 5.18 Arch Linux
    Memory: 12.14 GB / 23.23 GB
    Container: Yes
    Shell: 5.9 - /bin/zsh
  Binaries:
    Node: 16.15.1 - ~/.nvm/versions/node/v16.15.1/bin/node
    Yarn: 1.22.15 - ~/.nvm/versions/node/v16.15.1/bin/yarn
    npm: 8.11.0 - ~/.nvm/versions/node/v16.15.1/bin/npm
  Browsers:
    Chromium: 104.0.5112.79
    Firefox: 103.0.1
  npmPackages:
    vite: ^3.0.4 => 3.0.4
```


### Used Package Manager

yarn

### Logs

```
src/main.ts:4:15 - error TS2571: Object is of type 'unknown'.

4 const error = (await moduleImports[0]()).default;
```

### Validations

- [X] Follow our [Code of Conduct](https://github.com/vitejs/vite/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guidelines](https://github.com/vitejs/vite/blob/main/CONTRIBUTING.md).
- [X] Read the [docs](https://vitejs.dev/guide).
- [X] Check that there isn't [already an issue](https://github.com/vitejs/vite/issues) that reports the same bug to avoid creating a duplicate.
- [X] Make sure this is a Vite issue and not a framework-specific issue. For example, if it's a Vue SFC related bug, it should likely be reported to [vuejs/core](https://github.com/vuejs/core) instead.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/vitejs/vite/discussions) or join our [Discord Chat Server](https://chat.vitejs.dev/).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

--- a/packages/vite/types/importGlob.d.ts
+++ b/packages/vite/types/importGlob.d.ts
@@ -64,7 +64,7 @@ export interface ImportGlobFunction {
   /**
    * Import a list of files with a glob pattern.
    *
-   * Overload: 3 Module generic provided, infer the type from `eager: true`
+   * Overload 3: Module generic provided, infer the type from `eager: true`
    */
   <M>(
     glob: string | string[],

--- a/packages/vite/types/importGlob.d.ts
+++ b/packages/vite/types/importGlob.d.ts
@@ -38,7 +38,9 @@ export interface KnownAsTypeMap {
 
 export interface ImportGlobFunction {
   /**
-   * 1. No generic provided, infer the type from `eager` and `as`
+   * Import a list of files with a glob pattern.
+   *
+   * Overload 1: No generic provided, infer the type from `eager` and `as`
    */
   <
     Eager extends boolean,
@@ -51,14 +53,18 @@ export interface ImportGlobFunction {
     ? Record<string, T>
     : Record<string, () => Promise<T>>
   /**
-   * 2. Module generic provided, infer the type from `eager: false`
+   * Import a list of files with a glob pattern.
+   *
+   * Overload 2: Module generic provided, infer the type from `eager: false`
    */
   <M>(
     glob: string | string[],
     options?: ImportGlobOptions<false, string>
   ): Record<string, () => Promise<M>>
   /**
-   * 3. Module generic provided, infer the type from `eager: true`
+   * Import a list of files with a glob pattern.
+   *
+   * Overload: 3 Module generic provided, infer the type from `eager: true`
    */
   <M>(
     glob: string | string[],
@@ -68,7 +74,9 @@ export interface ImportGlobFunction {
 
 export interface ImportGlobEagerFunction {
   /**
-   * 1. No generic provided, infer the type from `as`
+   * Eagerly import a list of files with a glob pattern.
+   *
+   * Overload 1: No generic provided, infer the type from `as`
    */
   <
     As extends string,
@@ -78,7 +86,9 @@ export interface ImportGlobEagerFunction {
     options?: Omit<ImportGlobOptions<boolean, As>, 'eager'>
   ): Record<string, T>
   /**
-   * 2. Module generic provided
+   * Eagerly import a list of files with a glob pattern.
+   *
+   * Overload 2: Module generic provided
    */
   <M>(
     glob: string | string[],

